clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -6861,13 +6861,16 @@ ProcessDeclAttribute(Sema &S, Scope *scope, Decl *D, const ParsedAttr &AL,
   // though they were unknown attributes.
   if (AL.getKind() == ParsedAttr::UnknownAttribute ||
       !AL.existsInTarget(S.Context.getTargetInfo())) {
-    S.Diag(AL.getLoc(),
-           AL.isRegularKeywordAttribute()
-               ? (unsigned)diag::err_keyword_not_supported_on_target
-           : AL.isDeclspecAttribute()
-               ? (unsigned)diag::warn_unhandled_ms_attribute_ignored
-               : (unsigned)diag::warn_unknown_attribute_ignored)
-        << AL << AL.getRange();
+    if (AL.isRegularKeywordAttribute() || AL.isDeclspecAttribute()) {
```

Comment:
I think we may want a broader change that this; this prints the attribute namespace only for unknown attribute diagnostics. But we might want to fix up `FormatASTNodeDiagnosticArgument` to be smarter about printing attributes. That handles the case where we have an `Attr *` being passed to the diagnostics engine, but we may want to add a new "kind" to represent a `ParsedAttr` and update `const StreamingDiagnostic &operator<<(const StreamingDiagnostic &DB, const ParsedAttr &At)` to emit that new kind so the formatter can handle `ParsedAttr` similar to `Attr` in terms of printing.

I think that might be a bigger change though due to test churn, so perhaps that can be done separately as this is forward progress. CC @ErichKeane for additional opinions.

Reply:
I think that is a nice idea!  I think we actually already HAVE a ParsedAttr streaming diagnostic FWIW, its for `Attr *` (though we should probably make it work with `AttributeCommonInfo` instead, which works with `ParsedAttr` and `Attr`).

You're right about that being a bigger change, so I'm totally OK with that being a followup or a bug report.